backend/admin_digest_router.py

```
@@ -0,0 +1,141 @@
+        raise HTTPException(status_code=401, detail="digest authentication failed")
+
+    payload = _build_payload_or_500()
+
+    if NotificationDispatcher is None:
+        raise HTTPException(status_code=503, detail="telegram dispatcher not available")
+    try:
+        dispatcher = NotificationDispatcher()
+        await dispatcher.send_daily_digest(payload)
+    except Exception as exc:
+        log.exception("admin_digest.dispatch_failed exc=%s", exc)
+        raise HTTPException(status_code=502, detail=f"telegram dispatch failed: {exc}")
```

Ruff (0.15.17)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)